Folder iterator performance optimization

k8n opened the pull request:
See [this comment](https://github.com/sookasa/box.py/commit/8f59c68e3b60d949b5eab78917d4030946df9815#commitcomment-5748894)

I'd like to say that a test is forthcoming :-)

k8n committed:
Folder iterator performance optimization

See (this comment)[https://github.com/sookasa/box.py/commit/8f59c68e3b60d949b5eab78917d4030946df9815#commitcomment-5748894]

tals commented:
Your code has a bug - if content['entries'] is empty, it will be stuck in a loop.

k8n commented:
Sorry, tals. Drive-by coding failed on my part. Hopefully I got it right this time.

tals commented on box/client.py:
```diff
@@ -418,7 +418,10 @@ def get_folder_iterator(self, folder_id):
                 yield entry
-
+             
+            if  offset + len(content['entries']) >= content['total_count']:
```
nitpicking, but - can you put a comment there to explain the condition&break?
It makes parsing through code more pleasant :)

If not I can just add it later

tals commented on tests/test_client.py:
```diff
@@ -228,16 +228,43 @@ def test_get_folder_iterator(self):
             .once())
 
-        self.assertListEqual(list(client.get_folder_iterator(666)), list(range(10)))
+        self.assertListEqual(list(client.get_folder_iterator(666)), list(range(1001)))
```
switch to assertSequenceEqual() instead of forcing lists

k8n commented:
Cleaned up. Rolled-up changes into a single commit. Added requested changes. (Not going to try patch software working off an iPad in bed anymore :-). Thanks for your guidance.

k8n commented:
Merge me? I want to kill my fork.
